File: src/2025-02-23-deepclaude-try/cpp/exercises/practice/parallel-letter-frequency/parallel_letter_frequency.cpp
#include "parallel_letter_frequency.h"
#include <execution>
#include <algorithm>
#include <cctype>

namespace parallel_letter_frequency {
    std::unordered_map<char, int> frequency(const std::vector<std::string>& texts) {
        return std::transform_reduce(
            std::execution::par_unseq,
            texts.begin(), texts.end(),
            std::unordered_map<char, int>{},
            [](auto&& a, auto&& b) {
                for (const auto& [key, value] : b) {
                    a[key] += value;
                }
                return a;
            },
            [](const std::string& text) {
                std::unordered_map<char, int> counts;
                for (char c : text) {
                    if (std::isalpha(c)) {
                        counts[std::tolower(c)]++;
                    }
                }
                return counts;
            }
        );
    }
}

File: src/2025-02-23-deepclaude-try/javascript/exercises/practice/parallel-letter-frequency/parallel-letter-frequency.js
export const parallelLetterFrequency = async (texts) => {
  // Create a promise for processing each text concurrently
  const promises = texts.map(text => 
    new Promise(resolve => {
      const freq = {};
      // Count letter frequencies in this text
      for (const char of text.toLowerCase()) {
        if (/\p{L}/u.test(char)) {
          freq[char] = (freq[char] || 0) + 1;
        }
      }
      resolve(freq);
    })
  );
  
  // Wait for all texts to be processed
  const results = await Promise.all(promises);
  
  // Combine all frequency counts into a single result
  return results.reduce((total, freq) => {
    for (const [char, count] of Object.entries(freq)) {
      total[char] = (total[char] || 0) + count;
    }
    return total;
  }, {});
};
